### PR TITLE
[IMP] l10n_it_edi: Adding negative bill test

### PR DESCRIPTION
**Pull request by lordkrandel**
Added a test to verify the behaviour when receiving a negative bill. The invoice gets imported with negative amounts, but it can't be posted. When the user tries to post it, they are prompted to turn it into a credit note with a UserError.

### CODE REVIEW

**Review by jco-odoo — COMMENTED** [review]
Is not this generic behaviour anyways to raise an error?  The only thing specific to Italy is that we import it as negative.

File: addons/l10n_it_edi/tests/import_xmls/IT01234567890_FPR02.xml [hunk 1]
```diff
@@ -0,0 +1,117 @@
...
+    </FatturaElettronicaHeader>
+    <FatturaElettronicaBody>
+        <DatiGenerali>
+            <DatiGeneraliDocumento>
+                <TipoDocumento>TD01</TipoDocumento>
+                <Divisa>EUR</Divisa>
+                <Data>2014-12-18</Data>
+                <Numero>01234567890</Numero>
+                <Causale>LA FATTURA FA RIFERIMENTO AD UNA OPERAZIONE AAAA BBBBBBBBBBBBBBBBBB CCC DDDDDDDDDDDDDDD E
+                    FFFFFFFFFFFFFFFFFFFF GGGGGGGGGG HHHHHHH II LLLLLLLLLLLLLLLLL MMM NNNNN OO PPPPPPPPPPP QQQQ RRRR
```
[the remark this answers is not in the document]
Comment (lordkrandel) [reply addons/l10n_it_edi/tests/import_xmls/IT01234567890_FPR02.xml:59]: It's the only official Italian Tax Agency test file. I didn't write it myself by hand.

[the remark this answers is not in the document]
Comment (lordkrandel) [reply addons/l10n_it_edi/tests/import_xmls/IT01234567890_FPR02.xml:59]: Can change though

**Comment by lordkrandel** [comment]
@jco-odoo removed the posting parts, simplified the causale.

**Comment by jco-odoo** [comment]
@lordkrandel Something is off with the commit msg.  And what is the point?  Negative vendor bills exist and they work, but we want to make sure to keep supporting the case?  It will be in draft and then the user can still switch the move_type?  (there is a server action for it)

**Comment by lordkrandel** [comment]
As this behavior is generally not known, future changes could break it. I can definitely see any Odooer blocking the import of negative invoices before creating the draft for any kind of reason. Regression testing and peculiar use-case documentation are totally valid reasons to have a test.
Commit message will be fixed.

**Comment by lordkrandel** [comment]
Done.

**Comment by jco-odoo** [comment]
@robodoo r+

**Pull request closed without merging**